server/index.js

```diff
@@ -60,6 +60,90 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart/', (req, res, next) => {
+  const { cartId } = req.session;
+  const sql = `
+      select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using("productId")
+  where "c"."cartId" = $1
+  `;
+  const params = [cartId];
+  db.query(sql, params)
+    .then(result => {
+      if (!req.session.cartId) { res.status(200).json([]); }
+      res.status(200).json(result.rows[0]);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/cart/', (req, res, next) => {
+  const { productId } = req.body;
+  if (productId < 0) {
+    return next(new ClientError('Product Id must be a positive number ', 400));
+  }
+  const sql = `
+    select "price"
+      from "products"
+     where "productId" = $1
+    `;
+  const values = [productId];
+  db.query(sql, values)
+    .then(resultPrice => {
+      if (resultPrice.rows.length === 0) throw new ClientError('Invalid product', 400);
+      const sql = `
+            insert into "carts" ("cartId", "createdAt")
+            values (default, default)
+            returning "cartId"
+            `;
+      return db.query(sql)
+        .then(cartIdResult => (
+          {
+            price: resultPrice.rows[0].price, cartId: cartIdResult.rows[0].cartId
+          })
+        );
```

Comment:
If the `req.session` already contains a `cartId`, we don't want to make a whole new cart. In that case, you just want to return that `cartId` and the `price` and skip inserting the cart.